core/pkg/opaprocessor/processorhandler.go

```diff
@@ -154,6 +155,11 @@ func (opap *OPAProcessor) processRule(rule *reporthandling.PolicyRule, fixedCont
 
 	postureControlInputs := opap.regoDependenciesData.GetFilteredPostureControlInputs(rule.ConfigInputs) // get store
 
+	clusterName := cautils.ClusterName
+	provider := cloudsupport.GetCloudProvider(clusterName)
```

Comment:
I thought you want to get the cloud provider from the describe

Reply:
Right - but I saw describe is determind by "provider" in the same way - and for now, I need only the cloud provider and not the whole clusterDescribe. In case we will need additional info, we will request for the ClusterDescribe resource.

Reply:
But if a user runs Kubescape with the `--cluster-name=my-cluster-name` you will not run some controls because of this.
You should at least extract the provider from the context and not from the name.

Reply:
Done.

Reply:
We can't take it from context name
we need to take it from discovery from API server

Reply:
@Bezbran @dwertent
I've made the change:
1. First attempting to get the cloud provider from gitversion of discovered API version.
2. IF didn't succeed, attempting to get it from context.

To consider:
a. I only tested it for gke in which the "gke" is included in the git version string.
b. In case of using context (=cluster name) - this will be benefitial, as long as the cluster doesn't include a provider string ("gke", "aks", "eks") that is different than the actual provider.

Reply:
Actually EKS as "eks" in the git version and the same goes for AKS, so I think it's just fine